### PR TITLE
Issue #226: Functional e2e NLP example

### PR DESCRIPTION
**Pull request by shaahji**
Issue #226: Functional e2e NLP example

* Implemented a new version of Kernel and the CustomOp to support
  output that matches the HuggingFace model's input without the need
  for intermediate python logic.
* Implemented a e2e tutorial for exporting and inferencing using the
  HuggingFace's QuestionAnswering model.

Known Issue: Python side doesn't have an implementation of Bert Decoder
and so the augmented model is only half-complete. At the time of
inferencing the HuggingFace tokenizer is used to decode the result back
to string.

### CODE REVIEW

File: onnxruntime_extensions/pnp/_nlp.py [hunk 1]
```diff
@@ -35,10 +33,25 @@ def bert_tokenize(ctx, input_names, output_names, container, operator_name=None,
...
+        outputs=((_dt.INT64, []), (_dt.INT64, []), (_dt.INT64, [])))
+def bert_tokenizer(ctx, input_names, output_names, container, operator_name=None, **kwargs):
+    return create_bert_tokenizer(ctx, 'BertTokenizer', input_names, output_names,
+                                 container, operator_name=operator_name, **kwargs)
+
+
+@schema(inputs=((_dt.STRING, []),),
+        outputs=((_dt.INT64, []), (_dt.INT64, []), (_dt.INT64, [])))
+def bert_tokenizer_v2(ctx, input_names, output_names, container, operator_name=None, **kwargs):
+    return create_bert_tokenizer(ctx, 'HfBertTokenizer',
```
Comment (wenbingl) [inline onnxruntime_extensions/pnp/_nlp.py:50]: Can we unify the name, either it starts with hf, or ends with v2?

**Review by wenbingl — COMMENTED** [review]
Besides the e2e script, can you also add a unit test for HfBertTokenizer?

File: operators/tokenizer/bert_tokenizer.cc [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 #include <utility>
 
-BertTokenizerVocab::BertTokenizerVocab(std::string vocab) : raw_vocab_(vocab) {
-  auto tokens = SplitString(raw_vocab_, "\n", true);
+BertTokenizerVocab::BertTokenizerVocab(std::string_view vocab) : raw_vocab_(vocab) {
```
Comment (wenbingl) [inline operators/tokenizer/bert_tokenizer.cc:5]: is string_view introduced in C++ 17? IIRC, the largest C++ standard used in onnxruntime is c++14 for some reasons. And the onnxruntime-extensions may need to be consistent with it.

File: onnxruntime_extensions/pnp/_nlp.py (hunk 1 above)
Comment (shaahji) [reply onnxruntime_extensions/pnp/_nlp.py]: Done!

**Review by natke — APPROVED** [review]
Approving the e2e example :shipit: 

Someone else should review the other changes.

**Comment by shaahji** [comment]
> Besides the e2e script, can you also add a unit test for HfBertTokenizer?

Yes, I have the test locally. Trying to get the CI build issues resolved before I introduce the tests. PR coming soon to update the workflow, followed with tests.

**Comment by shaahji** [comment]
> Besides the e2e script, can you also add a unit test for HfBertTokenizer?

#269

**Review by wenbingl — APPROVED** [review]

File: operators/tokenizer/bert_tokenizer.cc (hunk 2 above)
Comment (wenbingl) [reply operators/tokenizer/bert_tokenizer.cc:5]: @snnn , is C++ 17 already supported in onnxruntime now?

**Pull request merged**